src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -71,23 +72,34 @@ public Path which(String fn) {
         if (shellDecorator != null) {
             decorated = shellDecorator.decorate(decorated);
         }
+        // First item in the command is the executable. If it's given as absolute path, add quotes around it
```

Comment:
Can we change the API to use `List<String>` instead of `String[]`? That will make it far easier to manipulate.

Reply:
Array is convenient for varargs which is used heavily in these APIs. I think this is okay.

Reply:
Yeah it is alright for varargs, but that's only the input, we can just create a list from the input array. We do a lot of manipulation, and using a list is a lot easier. ProcessBuilder takes a list.